pandas/types/cast.py

```diff
@@ -527,8 +527,9 @@ def _astype_nansafe(arr, dtype, copy=True):
     elif (np.issubdtype(arr.dtype, np.floating) and
           np.issubdtype(dtype, np.integer)):
 
-        if np.isnan(arr).any():
-            raise ValueError('Cannot convert NA to integer')
+        if not np.isfinite(arr).all():
+            raise ValueError('Cannot convert non-finite values to integer')
```

Comment:
can you add between brackets after "non-finite" something like "(NA or inf)", then it is clearer what is meant